server/src/api/gitLab/gitLabIntegration.ts

```diff
@@ -0,0 +1,213 @@
+          {
+            grant_type: 'authorization_code',
+            redirect_uri: redirectUri,
+          },
+          (err, access_token, refresh_token, result) => {
+            if (err) {
+              if (err.statusCode === 401 && /token/i.test(err.data))
+                reject(new InvalidTokenError(err.data));
+              else if (err.statusCode === 400 && /grant/i.test(err.data))
+                reject(new InvalidGrantError(err.data));
+              else reject(err);
+            } else
+              resolve({
+                accessToken: access_token,
+                refreshToken: refresh_token,
+              });
+          },
+        );
+      }),
+    tokenRefresh: (refreshToken, roadmapId) =>
+      new Promise((resolve, reject) => {
+        const redirectUri = `${redirectBaseURL}?roadmapId=${roadmapId}&integrationName=gitLab`;
+
+        oauth2(
+          config.consumerkey,
+          config.privatekey,
+          `https://${config.host}`,
+        ).getOAuthAccessToken(
+          refreshToken,
+          {
+            grant_type: 'refresh_token',
+            redirect_uri: redirectUri,
+          },
+          (err, access_token, refresh_token, result) => {
+            if (err) {
+              if (err.statusCode === 401 && /token/i.test(err.data))
+                reject(new InvalidTokenError(err.data));
+              else if (err.statusCode === 400 && /grant/i.test(err.data))
+                reject(new InvalidGrantError(err.data));
+              else reject(err);
+            } else
+              resolve({
+                accessToken: access_token,
+                refreshToken: refresh_token,
+              });
+          },
+        );
+      }),
+  }),
+};
+
+class GitLabImporter implements IntegrationProvider {
+  private readonly projectId: string;
+  private readonly oauth2: OAuth2;
+  private readonly accessToken: string;
+  private lastUsedRedirectUri: string;
+
+  constructor(
+    { projectId, consumerkey, privatekey, host }: IntegrationConfig,
+    tokens: { accessToken: string },
+  ) {
+    this.projectId = projectId || '';
+    this.lastUsedRedirectUri = '';
+    this.accessToken = tokens.accessToken;
+    this.oauth2 = oauth2(consumerkey, privatekey, host);
+  }
+
+  setLastUsedRedirectUri(uri: string) {
+    this.lastUsedRedirectUri = uri;
+  }
+  getLastUsedRedirectUri() {
+    return this.lastUsedRedirectUri;
+  }
+
+  async boards() {
+    const response = await this.fetch<{ id: number; name: string }[]>('boards');
+    return (response as any[]).map((board) => ({
+      id: `${board.id}`,
+      name: board.name,
+    }));
+  }
+
+  // In GitLab, columns ("lists") can only be created out of labels.
+  // The columns themselves do have an id, but they are named after the label.
+  // Therefore, this returns the column id and the label's name.
+  async columns(boardId: string) {
+    const lists = await this.fetch<
+      Array<{ id: number; label: { name: string } }>
```

Comment:
```suggestion
     { id: number; label: { name: string } }[]
```
Could use consistently the same notation for array, unless there is some specific reason